amazon-security-lake/package.yaml

```diff
@@ -61,7 +61,14 @@ pipelines:
     definition: |
       subscribe "{{ inputs.topic }}"
       where @name == "ocsf.network_activity"
-      to_amazon_security_lake "{{ inputs.s3-uri }}{{ inputs.network-activity-source-name }}",
+      
+      let $s3_bucket = "{{ inputs.s3-uri }}"
+      if $s3_bucket.ends_with("/") {
+        let $s3_uri = $s3_bucket
+      } else {
+        let $s3_uri = $s3_bucket + "/"
+      }
+      to_amazon_security_lake $s3_uri + "{{ inputs.network-activity-source-name }}",
```

Comment:
That does not work.

Reply:
The `if … { let … } else { let … }` pattern doesn't work, because `let` always updates the binding, no matter where it's located. This should be equivalent to the following right now:

```
let $s3_bucket = "{{ inputs.s3-uri }}"
let $s3_uri = $s3_bucket
let $s3_uri = $s3_bucket + "/"

if $s3_bucket.ends_with("/") {
} else {
}
```

This can also be considered a defect in the current implementation of `let`. In any case, to work around this, use a single `let $s3_uri` with an inline `if-else` expression.

Reply:
Ah, makes sense, I was wondering why I was able to access the let binding outside the inner scope.

I refactored the package to use complete URLs now anyways: these are easier for the users since there's a dedicated button in AWS where you can copy the whole source url.